bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/handler/BoxHandler.java

```diff
@@ -35,4 +58,45 @@ public BoxHandler(Bridge bridge, HttpClient httpClient,
             AVMFritzDynamicCommandDescriptionProvider commandDescriptionProvider) {
         super(bridge, httpClient, commandDescriptionProvider);
     }
+
+    @Override
+    protected void initConnections() {
+        AVMFritzBoxConfiguration config = getConfigAs(AVMFritzBoxConfiguration.class);
+        if (this.callMonitor == null && (isLinked(AVMFritzBindingConstants.CHANNEL_CALL_ACTIVE)
+                || isLinked(AVMFritzBindingConstants.CHANNEL_CALL_INCOMING)
+                || isLinked(AVMFritzBindingConstants.CHANNEL_CALL_OUTGOING)
+                || isLinked(AVMFritzBindingConstants.CHANNEL_CALL_STATE))) {
+            this.callMonitor = new CallMonitor(config.ipAddress, this, scheduler);
+        }
+        if (this.connection == null) {
+            if (config.password != null) {
+                this.connection = new FritzAhaWebInterface(config, this, httpClient);
+                stopPolling();
+                startPolling();
+            } else {
+                if (!callChannelsLinked()) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "The 'password' parameter must be configured to use the AHA features.");
+                }
+            }
+        }
+    }
+
+    private boolean callChannelsLinked() {
+        return getThing().getChannels().stream()
+                .filter(c -> isLinked(c.getUID()) && CALL_CHANNELS.contains(c.getUID().getId())).count() > 0;
+    }
+
+    @Override
+    public void dispose() {
+        if (callMonitor != null) {
+            callMonitor.dispose();
+        }
+        super.dispose();
+    }
+
+    @Override
+    public void updateState(String channelID, State state) {
```

Comment:
I do not understand this? Why do you need to override it and just call the super method?

Reply:
Because we make the method public (while being protected in the super class), so that the CallMonitor thread can access it on the handler object.